apis/Google.Cloud.Diagnostics.Common/Google.Cloud.Diagnostics.Common/Trace/SimpleManagedTracer.cs

```diff
@@ -50,20 +51,24 @@ internal Span(SimpleManagedTracer tracer, TraceSpan traceSpan)
             /// <summary> Ends the current span.</summary>
             public void Dispose()
             {
-                lock (_mutex)
+                GaxPreconditions.CheckState(!Disposed(), "Span cannot be disposed twice.");
+                if (Interlocked.CompareExchange(ref _ticks, DateTime.UtcNow.Ticks, 0) == 0)
+                {
+                    TraceSpan.EndTime = Timestamp.FromDateTime(new DateTime((long)_ticks).ToUniversalTime());
```

Comment:
I think this comment may have fallen through the cracks (or maybe you're waiting to see what Jon says).